tests/test_serializer.py

```diff
@@ -30,3 +32,67 @@ def test_message_list(self):
     def test_message_dict(self):
         d = {'m': message.Ping()}
         self.equal_after_processing(d)
+
+    def test_message_sig(self):
+        """Signed message inside a signed message"""
+
+        concent_keys = cryptography.ECCx(None)
+        provider_keys = cryptography.ECCx(None)
+        requestor_keys = cryptography.ECCx(None)
+
+        task_to_compute = message.TaskToCompute()
+        ctd = message.ComputeTaskDef({'task_id': 20, })
+        task_to_compute.compute_task_def = ctd
+
+        # Dump TaskToCompute to make it signed
+        s_task_to_compute = shortcuts.dump(
+            task_to_compute,
+            requestor_keys.raw_privkey,
+            provider_keys.raw_pubkey,
+        )
+
+        # Load TaskToCompute back to its original format
+        task_to_compute = shortcuts.load(
+            s_task_to_compute,
+            provider_keys.raw_privkey,
+            requestor_keys.raw_pubkey,
+        )
+
+        first_sig = task_to_compute.sig
+        first_hash = task_to_compute.get_short_hash()
+
+        # Task TaskToCompute is verified correctly
+        cryptography.ecdsa_verify(
+            requestor_keys.raw_pubkey,
+            task_to_compute.sig,
+            task_to_compute.get_short_hash(),
+        )
+
+        force_report = message.ForceReportComputedTask()
+        force_report.task_to_compute = task_to_compute
+
+        s_force_report = shortcuts.dump(
+            force_report,
+            provider_keys.raw_privkey,
+            concent_keys.raw_pubkey,
+        )
+
+        force_report = shortcuts.load(
+            s_force_report,
+            concent_keys.raw_privkey,
+            provider_keys.raw_pubkey,
+        )
+
+        second_sig = force_report.task_to_compute.sig
+        second_hash = force_report.task_to_compute.get_short_hash()
+
+        self.assertEqual(first_sig, second_sig)
+        self.assertEqual(first_hash, second_hash)
+
+        # Now, attached TaskToCompute should still be verified using
+        # original key
+        cryptography.ecdsa_verify(
```

Comment:
Is it supposed to throw if it's not signed properly? Can't see any asserts so it's not clear. Can you either assert that an exception is not thrown or add a comment?

Reply:
Not throwing an exception in unittest is default behaviour, you can't assert that. :)
https://github.com/golemfactory/golem-messages/blob/master/golem_messages/cryptography.py#L96

Not sure what comment can I add. `ecdsa_verify` is really short and rather straightforward.

Reply:
What I meant is that without knowing (or looking into) the internals of `ecdsa_verify` you don't know that it throws on failure. For a person looking at the code it could be returning a boolean instead. So a comment saying that it will throw helps to understand that this isn't in fact no-op. Or doing something verbose like
```
thrown = False
try:
  foo()
except:
  thrown = True
assert not thrown
```